src/vs/workbench/api/common/extHostTerminalService.ts

```diff
@@ -966,3 +989,9 @@ function asTerminalIcon(iconPath?: vscode.Uri | { light: vscode.Uri; dark: vscod
 function asTerminalColor(color?: vscode.ThemeColor): ThemeColor | undefined {
 	return ThemeColor.isThemeColor(color) ? color as ThemeColor : undefined;
 }
+
+function convertMutator(mutator: IEnvironmentVariableMutator): vscode.EnvironmentVariableMutator {
+	const newMutator: vscode.EnvironmentVariableMutator = { ...mutator };
+	delete (newMutator as any).variable;
```

Comment:
this is a little confusing to me

Reply:
The way API is designed exposing `variable` to consumers in Mutator object is redundant: https://code.visualstudio.com/api/references/vscode-api#EnvironmentVariableCollection

However it is needed in the internal mutator object. Hence this part of code just deletes it for the object exposed.